sdk/identity/identity/src/credentials/workloadIdentityCredential.ts

```diff
@@ -56,21 +62,66 @@ export class WorkloadIdentityCredential implements TokenCredential {
     const workloadIdentityCredentialOptions = options ?? {};
     const tenantId = workloadIdentityCredentialOptions.tenantId || process.env.AZURE_TENANT_ID;
     const clientId = workloadIdentityCredentialOptions.clientId || process.env.AZURE_CLIENT_ID;
+
     this.federatedTokenFilePath =
-      workloadIdentityCredentialOptions.tokenFilePath || process.env.AZURE_FEDERATED_TOKEN_FILE;
+      workloadIdentityCredentialOptions?.tokenFilePath || process.env.AZURE_FEDERATED_TOKEN_FILE;
+    this.serviceConnectionId = workloadIdentityCredentialOptions.serviceConnectionId;
+    if (!tenantId || !clientId) {
+      throw new CredentialUnavailableError(
+        `${credentialName}: is Unavailable. clientId and tenantId are required parameters/ environment varables - AZURE_TENANT_ID, AZURE_CLIENT_ID`
+      );
+    }
     if (tenantId) {
       checkTenantId(logger, tenantId);
```

Comment:
nit: you could move outside the `if` and remove the `if` entirely since you are validating tenantId exists 👍